### PR TITLE
Make user able to logout

### LINKED ISSUE #46: Allow users to logout
Users should be able to log out of their account.

Add a `/logout` endpoint that logs a user out. This will probably require a new function like `setLoggedInUser` that unsets the "user" session key. Unfortunately, allowing us to unset a value will mean changing the type of the session store! Why? Sessions only provide two functions: a lookup function, and an *insert* function. There is no delete function! So, the way to "unset" a key is to set the value to `Nothing`. So we need our `SessionStore` to be not a `Map` from `Text` to `Text` (like it currently is) but a map from `Text` to `Maybe Text`.

Steps:

1. Add a route for "/logout" in `routes` in `Web.hs`. The order of routes matters -- it should be right below the "/login" route. This route should be a `POST`, but unlike "/login", it doesn't need any parameters.
2. Add a handler function for the route to `Web.hs`. The handler should be something like `logoutHandler` and it should have the type `Ctxt -> IO (Maybe Response)`. 
3. Add a function to `User/Model.hs` that calls a function that unsets the "user" session key.
4. Change the type of the session store to a `Map Text (Maybe Text)`.

### PR DESCRIPTION
**Pull request by astynax**
Fixes #46 .

After this PR user will be able to logout using "Logout" button on user page.

### CODE REVIEW

File: app/templates/users/show.tpl [hunk 1]
```diff
@@ -9,7 +9,11 @@
 
   <body>
 
-    <h1>Welcome, <username /></h1>
+    <h1>Welcome, <username />
+      <form style="display: inline-block;" action="/logout" method="post">
+        <input type="submit" value="Logout">
+      </form>
+    </h1>
```
Comment (emhoracek) [inline app/templates/users/show.tpl]: Only [phrasing content](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements) (and not block-level content like forms) is allowed in heading tags -- so maybe put the logout button outside the header tag for now. It won't look perfect but it'll be semantically correct. :)

File: app/src/Session.hs [hunk 2]
```diff
@@ -30,5 +31,5 @@ setInSession ctxt k v =
 getFromSession :: Ctxt -> Text -> IO (Maybe Text)
 getFromSession ctxt k =
   case lookupSession ctxt of
-    Just (getSession, _) -> getSession k
+    Just (getSession, _) -> join <$> getSession k
```
Comment (emhoracek) [inline app/src/Session.hs]: 👍 love it

**Comment by astynax** [comment]
@emhoracek good point! I done some edits to make code more beginner-friendly

**Comment by emhoracek** [comment]
@astynax Appreciate you taking the extra time to do this! I'm going to merge. Thanks!

**Pull request merged**